### ISSUE TITLE
Problems getting cyflash 1.06 to work on PSoC 4200

### ISSUE THREAD

**Issue by openpinballproject** [arachnidlabs/cyflash#4]
I have a CY8CKIT-049-42xx which I attempted to use cyflash to upgrade.  It initially gave an error that was BootloaderError:  Invalid checksum type: 0.  Looking at the cyacd file, there are 16 bit checksums at the end of each line.  Trying the simplest jury rig, (i.e. changing the checksum type to 1), gave the following error:

python -m cyflash --serial COM10 Gen2.cyacd
Initialising bootloader.
Traceback (most recent call last):
  File "c:\Python27\lib\runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "c:\Python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "build\bdist.win32\egg\cyflash__main__.py", line 2, in <module>
    bootload.main()
  File "build\bdist.win32\egg\cyflash\bootload.py", line 217, in main
    "Device app ID %d is different from local app ID %d. Flash anyway? (Y/N)"))
  File "build\bdist.win32\egg\cyflash\bootload.py", line 121, in bootload
    self.enter_bootloader(data)
  File "build\bdist.win32\egg\cyflash\bootload.py", line 146, in enter_bootloader
    silicon_id, silicon_rev, bootloader_version = self.session.enter_bootloader()
  File "build\bdist.win32\egg\cyflash\protocol.py", line 240, in enter_bootloader
    response = self.send(EnterBootloaderCommand())
  File "build\bdist.win32\egg\cyflash\protocol.py", line 235, in send
    return command.RESPONSE.decode(response, self.checksum_func)
  File "build\bdist.win32\egg\cyflash\protocol.py", line 79, in decode
    raise InvalidPacketError()
cyflash.protocol.InvalidPacketError

I'm using version 1.06 of cyflash, 1.30 of the Cypress bootloader.  Before starting the cyflash program, I force the card into the bootloader while holding the button during boot.

**Comment by IronSean** [comment]
This seems a bit old, and at the time when you posted this the CyFlash tool only supported checksum type 1: CRC16. In Psoc Creator you can set a bootloader to use either Basic Summation checksums or CRC16. You would need to build the bootloader in CRC16 mode to use it at the time (just changing the value wouldn't change the actual checksums being calculated, and the packets would therefore read as invalid as their checksums would fail to match). 

Now the application supports both types, and should work in either mode.

**Comment by jsiverskog** [comment]
Can you please try with the current master?

**Comment by openpinballproject** [comment]
So I ran it with the current master.  Basically cloned the repository and ran the following commands:
cd into cloned repository:
c:\Python27\python.exe setup.py install
c:\Python27\python.exe -m cyflash --serial=COMx yyyyy.cyacd
where (x = COM port number of virtual COM port, and yyyyy = image name from PSoC creator)

All worked well on the PSoC 4200 that is sitting on my desk.  Newest version works well.  Thanks!

On comment from IronSean 2 years ago (never noticed that), I ended up pulling the repository and making modifications myself, and committing the updated version in my own repository to get it to work.  All attributions remained to give you guys the props since it is your code.  (Much better than trying to ferret through the incomplete documentation Cypress was providing at the time).  I also did not want to modify the bootloader in the PSoC 4200 from simple checksum to CRC16 for simple reason that it might brick the card.  (As it is, the Cypress bootloader at least on the PSoC 4200 is atrociously bloated).

Again, thanks for all the hard work,
-Hugh
